R/log_shiny_input_changes.R

```diff
@@ -37,35 +38,38 @@ log_shiny_input_changes <- function(
     input,
     namespace = NA_character_,
     excluded_inputs = character(),
-    excluded_pattern = "_width$") {
-  session <- shiny::getDefaultReactiveDomain()
-  if (!(shiny::isRunning() || inherits(session, "MockShinySession") || inherits(session, "session_proxy"))) {
-    stop("No Shiny app running, it makes no sense to call this function outside of a Shiny app")
+    excluded_pattern = "_width$",
+    session = shiny::getDefaultReactiveDomain()) {
+  stopifnot(inherits(input, "reactivevalues"))
+  stopifnot(is.character(namespace) && length(namespace) == 1)
+  stopifnot(is.character(excluded_inputs))
+  stopifnot(is.character(excluded_pattern) && length(excluded_pattern) <= 1)
```

Comment:
```suggestion
  stopifnot(is.character(excluded_pattern) && length(excluded_pattern) == 1)
```

string of length 0 breaks the code
```
r$> grepl(character(0L), "test")
Error in grepl(character(0L), "test") : invalid 'pattern' argument
```

Reply:
The docs already state `character(1)` so it does not require update